packages/integration-tests/utils/replayHelpers.ts

```diff
@@ -108,3 +120,60 @@ function getReplayPerformanceSpans(recordingEvents: RecordingEvent[]): Performan
     .filter(data => data.tag === 'performanceSpan')
     .map(data => data.payload) as PerformanceSpan[];
 }
+
+export function getFullRecordingSnapshots(replayRequest: Request): RecordingSnapshot[] {
+  const events = getDecompressedRecordingEvents(replayRequest) as RecordingEvent[];
+  return events.filter(event => event.type === 2).map(event => event.data as RecordingSnapshot);
+}
+
+function getDecompressedRecordingEvents(replayRequest: Request): RecordingEvent[] {
+  return replayEnvelopeRequestParser(replayRequest, 5) as RecordingEvent[];
+}
+
+/**
+ * Copy of the envelopeParser from ./helpers.ts, but with the ability
+ * to decompress zlib-compressed envelope payloads which we need to inspect for replay recordings.
+ * This parser can handle uncompressed as well as compressed replay recordings.
+ */
+const replayEnvelopeRequestParser = (request: Request | null, envelopeIndex = 2): Event => {
+  const envelope = replayEnvelopeParser(request);
+  return envelope[envelopeIndex] as Event;
+};
+
+const replayEnvelopeParser = (request: Request | null): unknown[] => {
+  // https://develop.sentry.dev/sdk/envelopes/
+  const envelopeBytes = request?.postDataBuffer() || '';
+
+  // first, we convert the bugger to string to split and go through the uncompressed lines
+  const envelopeString = envelopeBytes.toString();
```

Comment:
I wonder if we'll hit some string encoding issues -- does this need to be utf8? I guess easy way to test this is to add some unicode to our tests :)

Reply:
Good point. I added it to the improvement issue. I'm gonna tackle this separately to not widen the scope of this PR.